### PR TITLE
samples: migrate code from googleapis/java-retail

### PR DESCRIPTION
**Pull request by Shabirmean**
### Migrating samples from [googleapis/java-retail](https://github.com/googleapis/java-retail/tree/main/samples) into [java-docs-samples/retail](https://github.com/GoogleCloudPlatform/java-docs-samples)
---

- chore(samples): Code samples for Retail Tutorials. Search service samples (part 2) (#290)
- chore(samples): Retail Tutorials. CRUD products (#301)
- deps: update dependency com.google.cloud:google-cloud-bigquery to v2.9.3 (#322)
- deps: update dependency com.google.cloud:google-cloud-storage to v2.4.4 (#323)
- build(deps): update dependency org.codehaus.mojo:exec-maven-plugin to v1.6.0 (#321)
- build(deps): update dependency org.codehaus.mojo:exec-maven-plugin to v3 (#325)
- deps: update dependency com.google.code.gson:gson to v2.9.0 (#324)
- chore(samples): Retail Tutorials. Product Setup/Cleanup test resources (#291)
- chore(samples): Retail Tutorials. Import products (#297)
- chore(samples): Code samples for Retail Tutorials. Search service samples (part 1) (#289)
- deps: update dependency com.google.cloud:google-cloud-bigquery to v2.9.4 (#351)
- deps: update dependency com.google.cloud:google-cloud-storage to v2.4.5 (#329)
- chore(deps): update dependency com.google.cloud:google-cloud-retail to v2.0.12 (#318)
- chore(deps): update dependency com.google.cloud:google-cloud-retail to v2.0.13 (#360)
- chore(samples): Retail Tutorials. Events (write, rejoin, purge) (#303)
- deps: update dependency com.google.cloud:google-cloud-bigquery to v2.10.0 (#361)
- chore(deps): update dependency com.google.cloud:libraries-bom to v24.4.0 (#362)
- chore(deps): update dependency com.google.cloud:libraries-bom to v25 (#363)
- chore(samples): Retail Tutorials. Events import (#302)
- chore(deps): update dependency com.google.cloud:google-cloud-retail to v2.0.14 (#368)
- deps: update dependency com.google.cloud:google-cloud-bigquery to v2.10.1 (#370)
- chore(deps): update dependency com.google.cloud:google-cloud-retail to v2.0.15 (#373)
- deps: update dependency com.google.cloud:google-cloud-bigquery to v2.10.2 (#375)
- chore(samples): Major update according review comments. (#369)
- deps: update dependency com.google.cloud:google-cloud-storage to v2.5.0 (#378)
- deps: update dependency com.google.cloud:google-cloud-bigquery to v2.10.4 (#382)
- deps: update dependency com.google.cloud:google-cloud-storage to v2.5.1 (#385)
- chore(deps): update dependency com.google.cloud:google-cloud-retail to v2.0.16 (#380)
- chore(main): release 2.0.17 (#381)
- chore(deps): update dependency com.google.cloud:google-cloud-retail to v2.0.17 (#387)
- chore(deps): update dependency com.google.cloud:libraries-bom to v25.1.0 (#393)
- deps: update dependency com.google.cloud:google-cloud-bigquery to v2.10.5 (#394)
- deps: update dependency com.google.cloud:google-cloud-storage to v2.6.0 (#391)
- chore(deps): update dependency com.google.cloud:google-cloud-retail to v2.0.18 (#398)
- docs(samples): the bash scripts for environment setup are added (#392)
- deps: update dependency com.google.cloud:google-cloud-bigquery to v2.10.7 (#402)
- deps: update dependency com.google.cloud:google-cloud-bigquery to v2.10.8 (#403)
- deps: update dependency com.google.cloud:google-cloud-bigquery to v2.10.9 (#405)
- deps: update dependency com.google.cloud:google-cloud-storage to v2.6.1 (#406)
- deps: update dependency com.google.cloud:google-cloud-bigquery to v2.10.10 (#407)
- chore(deps): update dependency com.google.cloud:google-cloud-retail to v2.0.19 (#409)
- chore(deps): update dependency com.google.cloud:libraries-bom to v25.2.0 (#421)
- docs(samples): Retail. Prediction samples (#412)
- Revert "docs(samples): Retail. Prediction samples (#412)" (#430)
- deps: update dependency com.google.cloud:google-cloud-bigquery to v2.11.0 (#431)
- chore(deps): update dependency com.google.cloud:libraries-bom to v25.3.0 (#436)
- deps: update dependency com.google.cloud:google-cloud-bigquery to v2.11.1 (#437)
- deps: update dependency com.google.cloud:google-cloud-bigquery to v2.11.2 (#438)
- deps: update dependency com.google.cloud:google-cloud-storage to v2.7.1 (#444)
- chore(deps): update dependency com.google.cloud:google-cloud-retail to v2.1.0 (#445)
- deps: update dependency com.google.cloud:google-cloud-bigquery to v2.12.0 (#446)
- deps: update dependency com.google.cloud:google-cloud-storage to v2.7.2 (#447)
- chore(deps): update dependency com.google.cloud:libraries-bom to v25.4.0 (#450)
- deps: update dependency com.google.cloud:google-cloud-bigquery to v2.13.1 (#448)
- chore(deps): update dependency com.google.cloud:google-cloud-retail to v2.2.0 (#455)
- deps: update dependency com.google.cloud:google-cloud-storage to v2.8.0 (#456)
- deps: update dependency com.google.cloud:google-cloud-bigquery to v2.13.2 (#454)
- chore(deps): update dependency com.google.cloud:google-cloud-retail to v2.2.1 (#463)
- deps: update dependency com.google.cloud:google-cloud-bigquery to v2.13.3 (#464)
- deps: update dependency com.google.cloud:google-cloud-bigquery to v2.13.4 (#467)
- chore(deps): update dependency com.google.cloud:libraries-bom to v26 (#485)
- docs(samples): Refactoring search package (#415)
- docs(samples): Refactoring product package (CRUD) (#417)
- build(deps): update dependency org.codehaus.mojo:exec-maven-plugin to v3.1.0 (#491)
- chore(deps): update dependency com.google.cloud:google-cloud-retail to v2.3.0 (#502)
- deps: update dependency com.google.cloud:google-cloud-storage to v2.10.0 (#504)
- deps: update dependency com.google.code.gson:gson to v2.9.1 (#500)
- deps: update dependency com.google.cloud:google-cloud-storage to v2.11.0 (#507)
- deps: update dependency com.google.cloud:google-cloud-storage to v2.11.1 (#509)
- deps: update dependency com.google.cloud:google-cloud-storage to v2.11.2 (#510)
- chore (samples) Added a place for updateMask insertion from tutorials (#498)
- chore (samples) Warn user about fails on user_import_data_to_catalog.sh (#489)
- chore (samples) Warn user about fails on user_environment_setup.sh. (#490)
- chore (samples) Refactored error handling check for import product from GCS. (#505)
- fix(samples) Add fulfillment places code sample: didn't work (#488)
- fix (samples) Import user events BQ, missing UpdateUserEventsJson.java (#494)
- deps: update dependency com.google.cloud:google-cloud-bigquery to v2.14.5 (#503)
- chore(deps): update dependency com.google.cloud:google-cloud-retail to v2.4.0 (#514)
- chore(deps): update dependency com.google.cloud:libraries-bom to v26.1.0 (#516)
- chore (samples) Refactoring: added invalid catalog name in import product from gcs. (#506)
- fix (samples) Warning about wrong relativePath for parent (#508)
- deps: update dependency com.google.cloud:google-cloud-storage to v2.11.3 (#511)
- deps: update dependency com.google.cloud:google-cloud-bigquery to v2.14.6 (#519)
- deps: update dependency com.google.cloud:google-cloud-bigquery to v2.14.7 (#523)
- deps: update dependency com.google.cloud:google-cloud-bigquery to v2.15.0 (#524)
- chore(deps): update dependency com.google.cloud:libraries-bom to v26.1.1 (#525)
- fix(samples): removed region tags (#531)
- fix(samples): removed add time from lro. (#530)
- docs(samples): Refactoring events package (#481)
- docs(samples): Refactoring product package (#480)
- deps: update dependency com.google.cloud:google-cloud-bigquery to v2.16.0 (#535)
- fix(samples): removed env variables and buckets from creating bq. (#526)
- fix(samples): updating both json files. (#540)
- chore(deps): update dependency com.google.cloud:google-cloud-retail to v2.5.0 (#545)
- deps: update dependency com.google.cloud:google-cloud-bigquery to v2.16.1 (#542)
- deps: update dependency com.google.cloud:google-cloud-storage to v2.12.0 (#543)
- chore(deps): update dependency com.google.cloud:libraries-bom to v26.1.2 (#546)
- fix(samples): replaced your-project-id with default. (#541)
- fix (samples) user environment setup script. (#534)
- fix(samples): lro added out-of-order step. (#539)
- chore(samples): improved message if no result returned in search tutorials. (#537)
- fix(samples): bug fix for setup script (#549)
- fix(samples): bug fix for removing buckets from bq. (#548)
- fix(samples): Removed delays on LRO (#527)
- chore(samples): refactoring comments. (#528)
- samples(fix): bug fix, wrong projectId. (#575)
- chore(deps): update dependency com.google.cloud:google-cloud-retail to v2.5.1 (#580)
- deps: update dependency com.google.cloud:google-cloud-bigquery to v2.17.0 (#578)
- deps: update dependency com.google.cloud:google-cloud-storage to v2.13.0 (#581)
- chore(deps): update dependency com.google.cloud:google-cloud-retail to v2.5.2 (#584)
- chore(deps): update dependency com.google.cloud:libraries-bom to v26.1.3 (#585)

Fixes #issue

> It's a good idea to open an issue first for discussion.

- [ ] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] API's need to be enabled to test (tell us)
- [ ] Environment Variables need to be set (ask us to set them)
- [ ] **Tests** pass:   `mvn clean verify` **required**
- [ ] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] Please **merge** this PR for me once it is approved.

### CODE REVIEW

**Comment by Shabirmean** [comment]
@chenlei1216 - you can find my manual updates in the following commits: https://github.com/GoogleCloudPlatform/java-docs-samples/pull/7403/commits/8bf47af23639ae29b203d6281bc3eeb8439f72b5 and https://github.com/GoogleCloudPlatform/java-docs-samples/pull/7403/commits/756443745bc5e305ff37229045391c7fd9c80297

**Review by averikitsch — APPROVED** [review]

**Review by minherz — APPROVED** [review]
lgtm

**Review by lesv — COMMENTED** [review]
Why is tutorial1.img empty?

**Review by lesv — APPROVED** [review]
* retail/interactive-tutorials/images/tutorail1.img is empty.

I'd have a bunch of questions / comments if this was new code, but since it's legacy that's moving LGTM.

**Comment by lesv** [comment]
Let me know if you need me to merge this.

**Comment by Shabirmean** [comment]
> Let me know if you need me to merge this.

Thank you @lesv - Im waiting for someone from @GoogleCloudPlatform/cloud-retail-team to also TAL!

File: retail/interactive-tutorials/src/main/java/setup/SetupCleanup.java [hunk 1]
```diff
@@ -0,0 +1,407 @@
+/*
```
Comment (chenlei1216) [inline retail/interactive-tutorials/src/main/java/setup/SetupCleanup.java:1]: Just to double confirm: are both repositories using the same GCP project and environment variables for the sample tests? If not we may need to set them up in Cloud Console of the new projects following https://cloud.google.com/retail/docs/setting-up, and run the scripts to import the testing data.

**Review by chenlei1216 — APPROVED** [review]

File: retail/interactive-tutorials/src/main/java/setup/SetupCleanup.java (hunk 1 above)
Comment (Shabirmean) [reply retail/interactive-tutorials/src/main/java/setup/SetupCleanup.java:1]: @averikitsch can you help me answer this question?

Comment (Shabirmean) [reply retail/interactive-tutorials/src/main/java/setup/SetupCleanup.java:1]: This repo is using `java-docs-samples-testing` project according this setup:
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.kokoro/tests/run_tests.sh#L58

And the `java-retail` repo is using the same project for the [samples - presubmit](https://github.com/googleapis/java-retail/blob/main/.kokoro/presubmit/samples.cfg#L17).  

Another project `gcloud-devel` is also used for [some other pre-submit tests](https://github.com/googleapis/java-retail/blob/main/.kokoro/presubmit/integration.cfg#L22). But not sure if that impacts this

Comment (averikitsch) [reply retail/interactive-tutorials/src/main/java/setup/SetupCleanup.java:1]: The other pre-submit tests (integration) are library specific and are separate from the sample. The sample is running on the same test project.

Comment (Shabirmean) [reply retail/interactive-tutorials/src/main/java/setup/SetupCleanup.java:1]: Thank you Averi!

**Pull request merged**